R/coerce_to_messydate.R

```diff
@@ -120,30 +203,177 @@ standardise_widths <- function(dates) {
+                  st_negative(dates), dates)
+}
+
+as_ac_dates <- function(dates) {
+  # remove after christ letters
+  dates <- stringr::str_remove_all(dates, "(ad|AD|Ad|aD)")
+  dates <- stringr::str_trim(dates, side = "both")
+}
+
+st_negative_range <- function(dates) {
+  dates <- stringr::str_remove_all(dates, "(bc|BC|Bc|bC)")
+  dates <- gsub(" ", "", dates)
+  dates <- paste0("-", strsplit(dates, "\\.\\.")[[1]][1],
+                  "..-", strsplit(dates, "\\.\\.")[[1]][2])
+}
+
+st_negative_sets <- function(dates) {
+  dates <- stringr::str_remove_all(dates, "(bc|BC|Bc|bC)")
+  dates <- gsub(" ", "", dates)
+  dates <- unlist(strsplit(dates, "\\,"))
+  dates <- ifelse(length(dates) > 1,
+                  paste0("-", paste(dates, collapse = ", -")),
+                  paste0("-", dates))
+}
+
+st_negative <- function(dates) {
+  dates <- stringr::str_remove_all(dates, "(bc|BC|Bc|bC)")
+  dates <- stringr::str_trim(dates, side = "both")
+  dates <- paste0("-", dates)
+}
+
+ask_user <- function(dates) {
+  dates <- ifelse(stringr::str_detect(dates, "^([:digit:]{2})-([:digit:]{2})-([:digit:]{2})$") &
+                    as.numeric(gsub("-", "", stringr::str_extract(dates, "^[:digit:]{2}-"))) < 32 &
+                    as.numeric(gsub("-", "", stringr::str_extract(dates, "-[:digit:]{2}-"))) < 32 &
+                    as.numeric(gsub("-", "", stringr::str_extract(dates, "[:digit:]{2}$"))) < 32,
+                  reorder_ambiguous(dates), dates)
+  input <- utils::menu(c("Yes", "No"),
+                       title = paste0("Would you like to complete ambiguous 6 digit dates?"))
+  if (input == 1) {
+    dates <- ifelse(stringr::str_detect(dates, "^([:digit:]{2})-([:digit:]{2})-([:digit:]{2})$") &
+                      as.numeric(gsub("-", "", stringr::str_extract(dates, "^[:digit:]{2}-"))) < 23,
+                    complete_ambiguous_20(dates), dates)
+    dates <- ifelse(stringr::str_detect(dates, "^([:digit:]{2})-([:digit:]{2})-([:digit:]{2})$") &
+                      as.numeric(gsub("-", "", stringr::str_extract(dates, "^[:digit:]{2}-"))) > 22,
+                    complete_ambiguous_19(dates), dates)
+  }
+  dates
+}
+
+reorder_ambiguous <- function(d) {
+  input <- utils::menu(c("YMD (Year-Month-Day)", "DMY (Day-Month-Year)",
+                         "MDY (Month-Day-Year)"),
+                       title = paste0("What should the component order of ambiguous 6 digit dates be
+                                      (e.g. ", d, ")?"))
+  if (input == 1) {
+    out <- d
+    message("Ambiguous 6 digit dates already in standard YMD format")
+  }
+  if (input == 2) {
+    out <- stringr::str_replace_all(d, "^([:digit:]{2})-([:digit:]{2})-([:digit:]{2})$", "\\3-\\2-\\1")
+    message("Ambiguous 6 digit dates have been changed to standard YMD format")
+  }
+  if (input == 3) {
+    out <- stringr::str_replace_all(d, "^([:digit:]{2})-([:digit:]{2})-([:digit:]{2})$", "\\3-\\1-\\2")
```

Comment:
Same as above, would you like us to assume century instead of asking so that we automatically change to 4-2-2 structure? Please let me know and I will make the respective changes. Thank you.